qemu/tests/qmp_command.py

```diff
@@ -19,167 +21,127 @@ def run(test, params, env):
     :param params: Dictionary with the test parameters
     :param env: Dictionary with test environmen.
     """
-    def check_result(qmp_o, output=None, exception_list=""):
+
+    def check_list(qmp_o, key, val=None, check_item_in_pair=True):
+        """
+        Check if the expect key, val are contained in QMP output qmp_o.
+
+        :param qmp_o: output of QMP command
+        :type qmp_o: list
+        :param key: expect result
+        :type key: str
+        :param val: expect result
+        :type val: str or None(if check_item_in_pair=False)
+        :param check_item_in_pair: If expect result is dict (True) or str (False)
+        :type check_item_in_pair: bool.
+
+        :return check result
+        :rtype: bool
+        """
+        for element in qmp_o:
+            if isinstance(element, dict):
+                if _check_dict(element, key, val, check_item_in_pair):
+                    return True
+            elif isinstance(element, list):
+                if check_list(element, key, val, check_item_in_pair):
+                    return True
+            elif element != '' and not check_item_in_pair:
+                if strict_match:
+                    if operator.eq(key, element):
+                        return True
+                else:
+                    if key in str(element):
+                        return True
+        return False
+
+    def _check_dict(dic, key, val=None, check_item_in_pair=True):
+        """
+        Check if the expect key, val are contained in QMP output dic.
+
+        :param dic: content of QMP command return value
+        :type dic: dict
+        :param key: expect result
+        :type key: str
+        :param val: expect result
+        :type val: str or None(if check_item_in_pair=False)
+        :param check_item_in_pair: If expect result is dict or str
+        :type check_item_in_pair: bool. Means expect result is dict or str.
+
+        :return check result
+        :rtype: bool
+        """
+        if key in dic and not check_item_in_pair:
+            return True
+        elif key in dic and val == dic[key]:
+            return True
+        else:
+            for value in dic.values():
+                if isinstance(value, dict):
+                    if _check_dict(value, key, val, check_item_in_pair):
+                        return True
```

Comment:
The current logic is to return True as long as one value passes the check, is it expected?

Reply:
Yes, it is.